src/expression.c

```diff
@@ -3696,7 +3696,7 @@ Expression *StringExp::semantic(Scope *sc)
             string = buffer.extractData();
             len = newlen;
             sz = 4;
-            //type = new TypeSArray(Type::tdchar, new IntegerExp(loc, len, Type::tindex));
+            //type = new TypeSArray(Type::tdchar, new IntegerExp(loc, len, Type::tsize_t));
```

Comment:
You should probably just delete these lines, they look like D1 string stuff to me.